### PR TITLE
Fix for elementwise_add_grad GetExpectedKernelType function

### PR DESCRIPTION
**Pull request by arlesniak**
### PR types

Bug fixes
### PR changes

Others
### Describe

Fix for wrong conditions between forward and backward in elementwise_add_grad op

### CODE REVIEW

**Comment by jczaja** [comment]
@arlesniak Could you please paste a shape from this op that this change to enable?

File: paddle/fluid/operators/elementwise/elementwise_op.h [hunk 1]
```diff
@@ -319,16 +319,7 @@ class ElementwiseOpGrad : public framework::OperatorWithKernel {
         ctx, framework::GradVarName("Out"));
 
 #ifdef PADDLE_WITH_MKLDNN
-    // If broadcasting is needed, use native implementation
```
Comment (jakpiase) [inline paddle/fluid/operators/elementwise/elementwise_op.h:322]: So checking these conditions is no longer needed?

Comment (jczaja) [reply paddle/fluid/operators/elementwise/elementwise_op.h:322]: Apparently it should work.  FWD kernel is more limited by how broadcasting can be done, but here in BWD it is just reduction so according to oneDNN manual it should work  with reduction on any axes. So GetExpectedKernelType of BWD should match FWD. And this is what this PR is doing.

**Review by jczaja — APPROVED** [review]
LGTM

**Review by jakpiase — APPROVED** [review]
LGTM

**Comment by arlesniak** [comment]
@Aganlengzi Could you please review and merge this PR?

**Review by Aganlengzi — APPROVED** [review]
LGTM

**Pull request merged**